tactics/tacticals.ml

```diff
@@ -467,6 +467,7 @@ module New = struct
   (* Check that holes in arguments have been resolved *)
 
   let check_evars env sigma extsigma origsigma =
+    let origevars = Evar.Map.domain (Evd.undefined_map origsigma) in
```

Comment:
I'm a bit worried about the performance cost of this, maybe a lazy should be used here